jsontableschema/validate.py

```diff
@@ -4,284 +4,110 @@
-                              'fields key.')
-
-        # each `fields` key in a `foreignKey` MUST be a string or array
-        if not all([isinstance(o.get('fields'), (compat.str, list))
-                   for o in schema['foreignKeys']]):
-            valid = False
-            errors.append('A JSON Table Schema foreignKey.fields type must '
-                              'be a string or an array.')
-
-        for fk in schema['foreignKeys']:
-
-            # ensure that `foreignKey.fields` match field names
-            if isinstance(fk.get('fields'), compat.str):
-                if fk.get('fields') not in [f['name'] for f in
-                                            schema['fields']]:
-                    valid = False
-                    errors.append('A JSON Table Schema foreignKey.fields '
-                                      'value must correspond with field names.')
-
-            else:
-                for field in fk.get('fields'):
-                    if not field in [f['name'] for f in
-                                     schema['fields']]:
-                        valid = False
-                        errors.append('A JSON Table Schema '
-                                          'foreignKey.fields value must '
-                                          'correspond with field names.')
-
-            # ensure that `foreignKey.reference` is present and is a hash
-            if not isinstance(fk.get('reference'), dict):
-                valid = False
-                errors.append('A JSON Table Schema foreignKey.reference '
-                                  'must be a hash.')
-
-            # ensure that `foreignKey.reference` has a `resource` key
-            if not 'resource' in fk.get('reference', {}):
-                valid = False
-                errors.append('A JSON Table Schema foreignKey.reference '
-                                  'must have a resource key.')
-
-            # ensure that `foreignKey.reference` has a `fields` key
-            if not 'fields' in fk.get('reference', {}):
-                valid = False
-                errors.append('A JSON Table Schema foreignKey.reference '
-                                  'must have a fields key.')
-
-            # ensure that `foreignKey.reference.fields`
-            # matches outer `fields`
-            if isinstance(fk.get('fields'), compat.str):
-                if not isinstance(fk['reference']['fields'], compat.str):
-                    valid = False
-                    errors.append('A JSON Table Schema '
-                                      'foreignKey.reference.fields must match '
-                                      'field names.')
-            else:
-                if not len(fk.get('fields')) == len(fk['reference']['fields']):
-                    valid = False
-                    errors.append('A JSON Table Schema must have a fields '
-                                      'key.')
-
-    return valid, errors
+def validate(schema):
```

Comment:
similar to json schema, this will raise the first exception it encouters, I did consider having a MultipleInvalid exception along the lines of voluptous, but left it similar to jsonschema